runtime.md

```diff
@@ -1,5 +1,47 @@
 # Runtime and Lifecycle
 
+## State
+
+The runtime state for a container is persisted on disk so that external tools can consume and act on this information.
+The runtime state is stored in a JSON encoded file.
+It is recommended that this file is stored in a temporary filesystem so that it can be removed on a system reboot.
+On Linux based systems the state information should be stored in `/run/oci/containers`.
+The directory structure for a container is `/run/oci/containers/<containerID>/state.json`.
+By providing a default location that container state is stored external applications can find all containers running on a system.
+
+* **version** (string) Version of the OCI specification used when creating the container.
+* **id** (string) ID is the container's ID.
+* **pid** (int) Pid is the ID of the main process within the container.
+* **root** (string) Root is the path to the container's bundle directory.
+
+The ID is provided in the state because hooks will be executed with the state as the payload.
+This allows the hook to perform clean and teardown logic after the runtime destroys it's own state.
+The root directory to the bundle is provided in the state so that consumers can find the container's configuration and rootfs.
+
+*Example*
+
+```json
+{
+    "id": "oci-container",
+    "pid": 4422,
+    "root": "/containers/redis"
+}
+```
+
+Linux systems add some platform specific information to the state.
+
+* **externalFds** Paths to the container's open file descriptors.
+
+*Example Linux*
+
+```json
+{
+    "externalFds": [
+        "/proc/33/fd/1"
```

Comment:
On Sat, Aug 29, 2015 at 05:34:30AM -0700, Andrew Vagin wrote:
> > This isn't a libcontainer spec ;).
> 
> I am not an author of this pull request;))

True ;).  I agree with @philips [1] that it would be good to drop
externalFds from this pull request, get clarification on the
platform-independent fields, and get that merged.  Then we can land
the externalFds via a follow-up pull request that just focuses on that
issue, explaining how it's supposed to work and linking to the CRIU
docs for caveats and additional details.

> I think your understanding of externalFds is incomplete…

Before this conversation it was nonexistent, so at least I'm moving in
the right direction ;).  Hopefully we can boil my confusions down into
docs that will help future newcomers avoid the same traps.

[1]: https://github.com/opencontainers/specs/pull/87#issuecomment-136007462